chain/client/src/chunk_validation.rs

```diff
@@ -285,69 +287,25 @@ fn pre_validate_chunk_state_witness(
             };
             let is_new_chunk = chunk.is_new_chunk(block.header().height());
             block_hash = *block.header().prev_hash();
+            if is_new_chunk {
+                prev_chunks_seen += 1;
+            }
             if prev_chunks_seen == 0 {
                 blocks_after_last_chunk.push(block);
             } else if prev_chunks_seen == 1 {
                 blocks_after_last_last_chunk.push(block);
             }
-            if is_new_chunk {
-                prev_chunks_seen += 1;
-            }
             if prev_chunks_seen == 2 {
```

Comment:
nit: now you can make this a part of loop condition: replace `loop` -> `while prev_chunks_seen < 2`

Reply:
I tried it out, but I think I actually like the `loop` version more. The logical flow is cleaner - first we get a new block, and then decide what to do. Either put it in one of the lists or break out of the loop.
OTOH usually having a `while` is better than `loop`...
Idk, to avoid bikeshedding I think I would just leave it as is ¯\_(ツ)_/¯